flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -287,6 +291,23 @@ public void onUnknownDeploymentsOf(
 
         log.info("Initializing job {} ({}).", jobName, jid);
 
+        if (jobMasterConfiguration.getConfiguration().get(JobManagerOptions.SCHEDULER_MODE)
+                == SchedulerExecutionMode.REACTIVE) {
+            Preconditions.checkState(
+                    slotPoolServiceSchedulerFactory.getSchedulerType()
+                            == JobManagerOptions.SchedulerType.Adaptive,
+                    "Expecting Adaptive Scheduler for reactive mode");
```

Comment:
I am wondering whether the `SchedulerExecutionMode` shouldn't influence which scheduler is chosen. That way we wouldn't need this precondition here.